src/qt3utils/applications/qt3scan/controller.py

```diff
@@ -200,7 +211,319 @@ def scan_image_rightclick_event(self, event: MouseEvent) -> None:
+
+        """
+        raw_counts_for_axis, wavelengths = (
+            self.scan_axis('x', self.xmin, self.xmax, self.step_size)
+        )
+        # raw_counts_for_axis is of shape (N steps, M spectrum size)
+        # wavelengths is of shape (M spectrum size,)
+        assert len(wavelengths) == raw_counts_for_axis.shape[-1]
+
+        # rehape raw_counts to
+        # (1, N, M)
+        raw_counts_for_axis = raw_counts_for_axis.reshape(1, len(raw_counts_for_axis), -1)
+
+        if self.hyper_spectral_raw_data is None:
+            self.hyper_spectral_raw_data = raw_counts_for_axis
+            self.logger.debug(f'Creating new hyperspectral array of shape: {self.hyper_spectral_raw_data.shape}')
+        else:
+            if self.hyper_spectral_raw_data.shape[-1] != raw_counts_for_axis.shape[-1]:
+                raise QT3Error("Inconsistent spectrum size obtained during scan_x! Check your hardware."
+                               f"expected shape[-1] {self.hyper_spectral_raw_data.shape[-1]}. found {raw_counts_for_axis.shape[-1]}")
+
+            self.hyper_spectral_raw_data = np.vstack((self.hyper_spectral_raw_data, raw_counts_for_axis))
+
+        if self.hyper_spectral_wavelengths is None:
+            self.hyper_spectral_wavelengths = wavelengths
+
+        if np.array_equal(self.hyper_spectral_wavelengths, wavelengths) is False:
+            raise QT3Error("Inconsistent wavelength array obtained during scan_x! Check your hardware.")
+
+    def scan_axis(self, axis, min, max, step_size) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Moves the microscope along the specified axis from min to max in steps of step_size.
+        Returns a tuple of two numpy arrays
+        The first numpy array is the raw spectrum from the scan in the shape
+        (N, M) where N is the number of positions along the axis and M
+        is the size of the spectrum
+        The second numpy array is an array of wavelength values for the spectrum of shape (M,)
+        """
+        spectrums_in_scan = []
+
+        # we use these to check the returned spectrum
+        # and wavelength array for consistency.
+        # we also currently do not support the
+        # values of the wavelengths changing for each position
+        # that is, the spectrometer must scan over the same set of wavelengths each time.
+        wavelength_array = None
+        initial_spectrum_size = None
+
+        self.position_controller.go_to_position(**{axis: min})
+        time.sleep(self.raster_line_pause)
+        for val in np.arange(min, max, step_size):
+            self.position_controller.go_to_position(**{axis: val})
+            measured_spectrum, measured_wavelengths = self.daq_controller.sample_spectrum()
+
+            if initial_spectrum_size is None:
+                initial_spectrum_size = len(measured_spectrum)
+            if wavelength_array is None:
+                wavelength_array = measured_wavelengths
+            try: 
```

Comment:
we don't need a try-block here

Reply:
Ha!!! I just read my comment two lines below. I realize that is confusing. I meant try-block elsewhere. I'll add a commit.

Reply:
I think let's ignore this for now. It's a little more complicated than I realized. The try-block might have to go somewhere else to preserve data. And it's currently not a directly problem that a user is having, so we can table it for now.

Reply:
Thanks for clarifying :)